apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx

```diff
@@ -0,0 +1,116 @@
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import TextareaWithMarkdownPreview from '@cdo/apps/lib/levelbuilder/TextareaWithMarkdownPreview';
+import SaveBar from '@cdo/apps/lib/levelbuilder/SaveBar';
+import {navigateToHref} from '@cdo/apps/utils';
+import $ from 'jquery';
+
+export default function ProgrammingExpressionEditor({
+  initialProgrammingExpression
+}) {
+  const [name, setName] = useState(initialProgrammingExpression.name);
+  const [shortDescription, setShortDescription] = useState(
+    initialProgrammingExpression.shortDescription
+  );
+  const [isSaving, setIsSaving] = useState(false);
+  const [lastUpdated, setLastUpdated] = useState(
+    initialProgrammingExpression.lastUpdated
+  );
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (isSaving) {
+      $.ajax({
```

Comment:
we're moving toward using the [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) over jQuery/AJAX

Reply:
moved over to fetch! the testing isn't quite as good as fetch is more annoying to work with, but I think the main stuff is tested!